### PR TITLE
CB-1521. Retrieve root certificate from ipa server

### PR DESCRIPTION
**Pull request by handavid**
/v1/freeipa/ca.crt?{environmentCrn} endpoint added to FreeIpaV1Endpoint.
This endpoint uses the FreeIpaClient to retrieve the root certificate
from the IPA server. The certificate is returned Base64 encoded. This
code re-formats the certificate into PEM format before returning it to
the caller.

### CODE REVIEW

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java [hunk 1]
```diff
@@ -34,12 +34,19 @@
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = FreeIpaOperationDescriptions.GET_BY_ENVID, produces = ContentType.JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
             nickname = "getFreeIpaByEnvironmentV1")
-    DescribeFreeIpaResponse describe(@QueryParam("environment") String environmentId);
+    DescribeFreeIpaResponse describe(@QueryParam("environment") String environmentCrn);
+
+    @GET
+    @Path("ca.crt")
+    @Produces(MediaType.APPLICATION_JSON)
```
Comment (handavid) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java]: I've got a question about this. The response isn't really json. It's more like text/plain, but we don't aren't currently configured to produce text/plain. Do we care about this? Should I configure the service to produce text/plain instead?

Comment (keyki) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java]: We do something similar to cluster certs. There are 2 options I think
* change it to text/plan as you said and just return the cert as base64
* take a look at https://github.com/hortonworks/cloudbreak/blob/master/core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/AutoscaleV4Endpoint.java#L84 here we return json and the json fields themselves contains the base64 encoded fields

Comment (keyki) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java]: Now this endpoint called ca.crt so in this case probably the 1st case would be better. For the 2nd I think we should also rename the endpoint to /certificates or something.

Comment (lacikaaa) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java]: do we need only the certificate or should we return something else in the response?

Comment (handavid) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java]: We only need the cert itself at the moment. If we find that we need more information, we can add a separate /certificates endpoint as @keyki suggests.

**Comment by handavid** [comment]
changed to return text/plain

**Pull request merged**